### PR TITLE
Feature/login api

### PR DESCRIPTION
**Pull request by miczoc98**
add jwt token refresh to login api

### CODE REVIEW

File: __tests__/qraphql_queries/user.test.js [hunk 1]
```diff
@@ -25,8 +25,8 @@ describe("test validity of GraphQL queries and mutations", () => {
 
     describe("register mutation", () => {
         const register_mutation = `
-            mutation Register ($user: String!, $pass: String!){
-                register(username: $user, password: $pass){
+            mutation Register ($user: String!, $email: String!, $pass: String!){
```
Comment (miczoc98) [inline __tests__/qraphql_queries/user.test.js:28]: fixed test broken due to addition of email in register mutation

File: __tests__/unit/auth/auth.test.js [hunk 2]
```diff
@@ -40,45 +41,51 @@ describe("extractTokenFromHeader function", () => {
 });
 
 describe("authenticateToken function", () => {
```
Comment (miczoc98) [inline __tests__/unit/auth/auth.test.js:43]: changed tests to fit with changes to function

File: front/src/components/AdminPanel/Panel/CompetitionDetails/CompetitionDetails.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import "./CompetitionDetails.css";
```
Comment (miczoc98) [inline front/src/components/AdminPanel/Panel/CompetitionDetails/CompetitionDetails.js:1]: auto-formatted during merge

File: front/src/components/AdminPanel/Panel/Competitors/Competitors.css [hunk 4]
```diff
@@ -1,55 +1,54 @@
-.table-div
```
Comment (miczoc98) [inline front/src/components/AdminPanel/Panel/Competitors/Competitors.css:1]: auto-formatted during merge

File: front/src/components/AdminPanel/Panel/Competitors/Competitors.js [hunk 5]
```diff
@@ -7,20 +7,20 @@ import competitors from "./dataCompetitors.json";
```
Comment (miczoc98) [inline front/src/components/AdminPanel/Panel/Competitors/Competitors.js:7]: auto-formatted during merge

File: front/src/components/AdminPanel/Panel/Competitors/Competitors.test.js [hunk 6]
```diff
@@ -1,72 +1,74 @@
-import { render, screen, within } from "@testing-library/react";
```
Comment (miczoc98) [inline front/src/components/AdminPanel/Panel/Competitors/Competitors.test.js:1]: auto-formatted during merge

File: front/src/components/AdminPanel/Panel/Competitors/dataCompetitors.json [hunk 7]
```diff
@@ -1,51 +1,50 @@
-[
```
Comment (miczoc98) [inline front/src/components/AdminPanel/Panel/Competitors/dataCompetitors.json:1]: auto-formatted during merge

File: front/src/components/AdminPanel/Panel/PanelLayout.css [hunk 8]
```diff
@@ -38,4 +38,4 @@
     .panelContentDiv {
```
Comment (miczoc98) [inline front/src/components/AdminPanel/Panel/PanelLayout.css:38]: auto-formatted during merge

File: __tests__/qraphql_queries/user.test.js [hunk 9]
```diff
@@ -25,8 +25,8 @@ describe("test validity of GraphQL queries and mutations", () => {
...
+                register(username: $user, email: $email, password: $pass){
```
Comment (miczoc98) [inline __tests__/qraphql_queries/user.test.js:29]: fixed test to account for changes in register mutation

Comment (frogi16) [reply __tests__/qraphql_queries/user.test.js:29]: Very good, I wanted to mention it during our next meeting 👍

**Review by frogi16 — APPROVED** [review]
I like to use "it" instead of "test" when test name begins with "should". It's more natural to read.
But this is a really, really minor thing and a matter of preference, so approved.

**Review by pieczonkam — APPROVED** [review]

**Pull request merged**